app/src/main/java/io/customer/example/MainActivity.kt

```diff
@@ -23,24 +24,70 @@ class MainActivity : AppCompatActivity() {
 
         // 2nd way
         makeAsynchronousRequest()
+
+        // log events
+//        makeEventsRequests()
+    }
+
+    private val outputCallback = Action.Callback<Unit> { result ->
+        when (result) {
+            is ErrorResult -> Log.v("ErrorResult", result.error.getDisplayMessage())
+            is Success -> Log.v("Success", "Success")
+        }
+    }
+
+    data class Fol(val a: String, val c: Int)
+
+    private fun makeEventsRequests() {
+        CustomerIo.instance().track(
+            name = "string event",
+            attributes = mapOf(
+                "value" to "string test",
+                "target" to 1
+            )
+        ).enqueue(outputCallback)
+        CustomerIo.instance().track(
+            name = "int event",
+            attributes = mapOf("value" to 1)
+        ).enqueue(outputCallback)
+        CustomerIo.instance().track(
+            name = "long event",
+            attributes = mapOf("value" to 1L)
+        ).enqueue(outputCallback)
+        CustomerIo.instance().track(
+            name = "array event",
+            attributes = mapOf("value" to listOf("1", "2"))
+        ).enqueue(outputCallback)
+        CustomerIo.instance().track(
+            name = "date event",
+            attributes = mapOf("value" to Date())
+        ).enqueue(outputCallback)
+        CustomerIo.instance().track(
+            name = "timestamp event",
+            attributes = mapOf("value" to Date().time)
+        ).enqueue(outputCallback)
+        CustomerIo.instance().track(
+            name = "custom class event",
+            attributes = mapOf("value" to Fol(a = "aa", c = 1))
+        )
     }
 
     private fun makeAsynchronousRequest() {
         CustomerIo.instance()
             .identify(
-                "identifier",
-                attributes = mapOf("email" to IdentityAttributeValue.StringAttribute("sample@email.com"))
-            ).enqueue {
-                when (it) {
-                    is ErrorResult -> Log.v("ErrorResult", it.error.cause.toString())
-                    is Success -> Log.v("Success", "Success")
-                }
-            }
+                identifier = "identifier",
+                attributes = mapOf("email" to "testemail@email.com")
+            ).enqueue(outputCallback)
     }
 
     private fun makeSynchronousRequest() {
         CoroutineScope(Dispatchers.IO).launch {
-            when (val result = CustomerIo.instance().identify("testcoo@email.com").execute()) {
+            when (
+                val result =
+                    CustomerIo.instance()
```

Comment:
I believe `CustomerIO` (all caps IO) is the more proper way for the name of the company.